neo4j-migrations-core/src/main/java/ac/simons/neo4j/migrations/core/DefaultCypherResource.java

```diff
@@ -165,16 +181,35 @@ private List<String> readStatements() {
 				if (statement.isEmpty()) {
 					continue;
 				}
-				newStatements.add(statement);
+				Matcher useMatcher = USE_DATABASE_PATTERN.matcher(statement);
+				boolean isMl = statement.contains("\n");
+				StringBuffer finalStatement = new StringBuffer();
```

Comment:
`StringBuffer` because of thread-safety?

Reply:
Because see 2nd commit. The appendReplacement overloads with a builder appeared in JDK 9.